hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieClientOnCopyOnWriteStorage.java

```diff
@@ -404,6 +406,38 @@ public void testPreCommitValidationWithMultipleInflights() throws Exception {
     assertTrue(testTable.commitExists(instant2));
   }
 
+  @Test
+  public void testInertsWithEmptyCommitsHavingWriterSchemaAsNull() throws Exception {
+    // Set autoCommit false
+    HoodieWriteConfig.Builder cfgBuilder = getConfigBuilder().withAutoCommit(false);
+    addConfigsForPopulateMetaFields(cfgBuilder, true);
+    SparkRDDWriteClient client = getHoodieWriteClient(cfgBuilder.build());
```

Comment:
Get the client in try-with-resource block, or close the client in finally block.

Reply:
> Get the client in try-with-resource block

This makes the client a final variable and re-init was not possible. Added the close in a finally block.